nexuse2e-core/src/main/java/org/nexuse2e/ui2/rest/DataManipulationHandler.java

```diff
@@ -86,7 +87,26 @@ private void stopMessages(HttpServletRequest request, HttpServletResponse respon
         }
     }
 
-    private void deleteConversations(HttpServletRequest request, HttpServletResponse response) {
-    }
+    private void deleteConversations(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String requestBody = readAll(request.getInputStream());
+        String[] conversationIds = new Gson().fromJson(requestBody, String[].class);
+        ArrayList<String> failedConversations = new ArrayList<>();
+
+        for (String conversationId : conversationIds) {
+            try {
+                ConversationPojo conversation = Engine.getInstance().getTransactionService().getConversation(conversationId);
+                Engine.getInstance().getTransactionService().deleteConversation(conversation);
+            } catch (NexusException e) {
+                failedConversations.add(conversationId);
+                LOG.error("An error occurred while trying to delete conversation " + conversationId, e);
+            }
+        }
 
+        if (failedConversations.isEmpty()) {
+            response.setStatus(HttpServletResponse.SC_OK);
+        } else {
+            response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+                    "An internal server error occurred while trying to delete conversations: " + failedConversations);
```

Comment:
It is possible that only some conversations are deleted? Should this mentioned in the error message?